provider.go

```diff
@@ -458,6 +461,17 @@ func (p *Provider) MountSecretsStoreObjectContent(ctx context.Context, attrib ma
 	return nil
 }
 
+func validateFilePath(path string) error {
+	segments := strings.Split(strings.ReplaceAll(path, `\`, "/"), "/")
```

Comment:
filepath.ToSlash might be more canonical for the replace?

Reply:
Yeah agreed - I did use that initially, but the fact that the behaviour of that function changes depending on the OS made it slightly more difficult to reason about and write cross-platform tests for:

```go
func ToSlash(path string) string {
	if Separator == '/' {
		return path
	}
	return strings.ReplaceAll(path, string(Separator), "/")
}
```

Using `strings.ReplaceAll` directly definitely makes the checking less correct, because (surprisingly to me) Linux/Unix file names can contain backslashes. However, I also wasn't 100% confident about whether mixing Windows and Linux was possible between pods and host node volumes, so decided to plump for the simpler fail-safe option of just treating all slashes as path separators. WDYT?

Reply:
Another benefit to using `strings.Replace` is that it can protect against strings such as  `"foo/.\b./bar"` which gets past `path.Join` as `"foo/../bar"` which then gets passed on to `ioutil.WriteFile` as such ([Playground example](https://play.golang.org/p/7unVBwRjhaS)).

Reply:
@tomhjp An interesting case where `strings.ReplaceAll` (or `filepath.ToSlash`) won't work: https://play.golang.org/p/BWcISJtKQpq

Reply:
Thanks @calvn, this is a really good idea to test!

After testing it out, I think this is ok, because while \b isn't printable, the character is still there, so if you actually create that directory, it will get created as `foo/.^H./bar/`, and not get interpreted as `foo/../bar`. It's slightly clearer to play around with on your own machine, but I think this also shows it: https://play.golang.org/p/6oktHKATJLH

Reply:
One separate thing this has made me realise is the code will currently fail for any paths specifying folders because the path doesn't exist. I'll add a call to `os.MkdirAll` to remedy.

Reply:
What's the use case for slashes? I don't think it matters since all secrets are forced in a certain directory.

Reply:
As discussed offline, it's more about handling keys with slashes gracefully. Because we do a write with the object name as the relative path, any slashes will get interpreted as path separators. We can either disallow all slashes in the object name, or implement a small feature to make them work - but what I wanted to avoid was just letting the inevitable error happen without handling it (when slashes are present), and therefore return a confusing error to the user.